docs/Guide/application-commands/interaction-handlers/what-are-they.mdx

```diff
@@ -3,7 +3,180 @@ sidebar_position: 1
+
+## How does it work?
+
+When creating your interaction handler, you have to specify the type it is. Currently, there are three types:
+  
+| Enum Name  | What it will run for |
+| ------------- | ------------- |
+| Autocomplete | Autocomplete for interaction commands  |
+| Button | Button presses  |
+| ModalSubmit | Pop up modals  |
+| MessageComponent | components  | Any message component interactions received (right now buttons, select menus, or modals)  |
+| SelectMenu | Drop-down menus in messages  |
+
+These types define what the handler should get. For instance, specifying a `Button` interaction handler type will make that handler receive **just** button interactions.
+
+But that's not all! There's also a `parse` method which can be used for even more granular control (read below).
+
+:::warning
+
+By default, there is no filter for the `parse` method! This means you'll get all interactions of that type, and if you want to override that (for example to only run that handler for certain ids), you'll want to override the `parse` method in your code!
+
+:::
+
+## The `parse` method
+
+When you're creating handlers, you may want to run certain handlers only when the custom id matches something you want. In order to allow filtering, as well as optional additional pre-parsing of the custom id, we provide the `parse` method on handlers, which you should override with your own code. The return of the `parse` method **must** be a `Some` / `None`, which you can do with the provided shortcuts in the class: `this.some()` and `this.none()`.
+
+:::info
+
+Returning a some means that the handler's run method should be called. Returning a none means this handler should be skipped for that interaction. If you've used our Args system, this might seem familiar with the ok/err that is present there. 
+
+:::
+
+What you return in the some is up to you! You can return nothing (be it null or undefined), a boolean, an object, a class, a box of hugs, you name it! But, if the handler should run, you must return a some (you can use the this.some() shortcut for it)
+
+:::caution
+
+If this function throws an error, the handler will be skipped and an `interactionHandlerParseError` event will be emitted!
+
+:::
+
+### Couldn't you shove everything in the parse method?
+
+The (sad) answer is... sort of.
+
+You should not shove everything in the parse method. It is meant strictly for filtering the custom id or the data received before actually working on it (for instance prefetching data from a database). It's done in this way to make you abstract the validation from the processing of the interaction. 
+
+## Example `parse` methods
+
+What you do in the parse method is up to you, however, we encourage keeping the parse method super simple and fast - remember, you must reply to interactions in **at max 3 seconds!**
```

Comment:
```suggestion
What you do in the parse method is up to you, however, we encourage keeping the parse method super simple and fast - remember, you must reply to interactions in **at max 3 seconds!**

With that said, if you know your interaction handler in total will take more than 3 seconds to complete, you should call the `deferReply` method in the `parse` method. That way, users will know your bot is processing instead of getting a `The application did not respond in time` message.
```